source/run-command/run-command.rst

```diff
@@ -187,7 +193,160 @@ Drivers MUST document the behavior of RunCommand if a ``maxTimeMS`` field  is al
 * See Client Side Operations Timeout's section on `runCommand <https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst#runcommand>`_
 * See Client Side Operations Timeout's section on `runCommand behavior <https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst#runcommand-behavior>`_
 
+
+--------------------
+``runCursorCommand``
+--------------------
+
+The following represents how a runCursorCommand API may be exposed.
+
+.. code:: typescript
+
+    interface Database {
+      /**
+       * Takes an argument representing an arbitrary BSON document and executes it against the server.
+       */
+      runCursorCommand(command: BSONDocument, options: RunCursorCommandOptions): RunCommandCursor;
+    }
+
+    interface RunCursorCommandOptions extends RunCommandOptions {
+      /**
+       * For operations that create cursors, timeoutMS can either cap the lifetime of the cursor or be applied separately to the original operation and all subsequent calls.
+       * To support both of these use cases, these operations MUST support a timeoutMode option.
+       *
+       * @defaultValue 'CURSOR_LIFETIME'
+       *
+       * @see https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst
+       */
+      timeoutMode?: 'ITERATION' | 'CURSOR_LIFETIME';
+
+      /**
+       * Identifies the type of cursor this is.
+       *
+       * A tailable cursor can receive empty `nextBatch` arrays in `getMore` responses.
+       * However, subsequent `getMore` operations may return documents if new data has become available.
+       *
+       * A tailableAwait cursor is an enhancement where instead of dealing with empty responses the server will block until data becomes available.
+       *
+       * @defaultValue NON_TAILABLE
+       */
+      cursorType: NON_TAILABLE | TAILABLE | TAILABLE_AWAIT;
```

Comment:
will you be adding tests for this?

Reply:
I have added coverage for tailable, awaitData is not something we can confirm without CSOT I think 🤔 but I'm also not too concerned with coverage, the enhancement is that the server blocks for the maxTimeMS, a driver's code doesn't change for the await case.

Reply:
I am mostly concerned with making sure that the options are set correctly on the cursor when it's created via this command

Reply:
Make sure we copy the CSOT tests for each cursorType as applied to the generic run cursor command